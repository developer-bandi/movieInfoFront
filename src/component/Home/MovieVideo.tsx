import ReactPlayer from 'react-player/lazy';
import styled from 'styled-components';

interface MovieVideoProps {
  moviekey: string | null;
}

const MovieVideo = ({ moviekey }: MovieVideoProps) => {
  return (
    <ReactPlayerBlock>
      {moviekey === 'default' ? (
        <Loading />
      ) : (
        <ReactPlayer
          url={`https://www.youtube.com/watch?v=${moviekey}`}
          loop={true}
          controls={true}
          playing={true}
          muted={true}
          width={'100%'}
          height={'100%'}
        />
      )}
    </ReactPlayerBlock>
  );
};

const ReactPlayerBlock = styled.section`
  height: 35rem;
  @media screen and (max-width: 500px) {
    height: 15rem;
  }
`;

const Loading = styled.div`
  background: black;
  height: 35rem;
  width: 100%;

  @media screen and (max-width: 500px) {
    height: 15rem;
  }
`;

export default MovieVideo;
