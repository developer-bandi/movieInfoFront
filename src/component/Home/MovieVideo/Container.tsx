import { useSelector } from 'react-redux';
import { ReducerType } from '../../../store';
import MovieVideoPresentational from './Presentational';

const MoiveVideoContainer = () => {
  const moviekey = useSelector((state: ReducerType) => state.homePoster);
  return <MovieVideoPresentational moviekey={moviekey} />;
};

export default MoiveVideoContainer;
