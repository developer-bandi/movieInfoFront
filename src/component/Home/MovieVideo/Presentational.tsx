import ReactPlayer from 'react-player/lazy';
import { HomePosterState } from '../../../store/homePoster/Reducer';
import styles from './Style';
interface MovieVideoProps {
  moviekey: HomePosterState;
}

const MovieVideoPresentational = ({ moviekey }: MovieVideoProps) => {
  if (moviekey.loading) {
    return <styles.Loading />;
  } else if (moviekey.error) {
    return (
      <styles.Error>
        <styles.ErrorMessage>에러가 발생하였습니다</styles.ErrorMessage>
      </styles.Error>
    );
  } else {
    return (
      <styles.MainBlock>
        <ReactPlayer
          url={`https://www.youtube.com/watch?v=${moviekey.content.posterList?.key}`}
          loop={true}
          controls={true}
          playing={true}
          muted={true}
          width={'100%'}
          height={'100%'}
        />
      </styles.MainBlock>
    );
  }
};

export default MovieVideoPresentational;
