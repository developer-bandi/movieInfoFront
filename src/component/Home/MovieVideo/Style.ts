import styled from 'styled-components';

const MainBlock = styled.section`
  height: 35rem;
  @media screen and (max-width: 500px) {
    height: 15rem;
  }
`;

const Loading = styled.div`
  background: black;
  height: 35rem;
  width: 100%;

  @media screen and (max-width: 500px) {
    height: 15rem;
  }
`;

const Error = styled.div`
  background: black;
  height: 35rem;
  width: 100%;
  display: flex;
  justify-content: center;
  align-items: center;
`;
const ErrorMessage = styled.div`
  font-size: 40px;
  color: white;
`;

const styles = {
  MainBlock,
  Loading,
  Error,
  ErrorMessage,
};

export default styles;
