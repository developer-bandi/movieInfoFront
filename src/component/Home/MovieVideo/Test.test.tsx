import { render, screen } from '@testing-library/react';
import MovieVideo from './Presentational';

describe('<MovieVideo/> Presentational', () => {
  const PosterInfo = {
    id: 1,
    title: 'test',
    overview: 'testOverview',
    voteAverage: 9.0,
    posterPath: 'testPath',
  };

  it('로딩중', () => {
    const utils = render(
      <MovieVideo
        moviekey={{
          content: {
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: true,
          error: false,
        }}
      />
    );
    expect(utils.container).toMatchSnapshot();
  });

  it('에러발생', () => {
    const utils = render(
      <MovieVideo
        moviekey={{
          content: {
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: true,
        }}
      />
    );
    expect(utils.container).toMatchSnapshot();
    screen.getByText('에러가 발생하였습니다');
  });

  it('정상 작동', () => {
    const utils = render(
      <MovieVideo
        moviekey={{
          content: {
            posterList: {
              nowShowingInfo: [PosterInfo],
              nowCommingInfo: [PosterInfo],
              key: '12345',
            },
            page: {
              nowShowingInfo: 1,
              nowCommingInfo: 1,
            },
          },
          loading: false,
          error: false,
        }}
      />
    );
    expect(utils.container).toMatchSnapshot();
  });
});
