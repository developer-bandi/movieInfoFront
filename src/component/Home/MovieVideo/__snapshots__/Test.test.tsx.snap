// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieVideo/> Presentational 로딩중 1`] = `
<div>
  <div
    class="sc-gsDKAQ coKEUB"
  />
</div>
`;

exports[`<MovieVideo/> Presentational 에러발생 1`] = `
<div>
  <div
    class="sc-dkPtRN bFvZKz"
  >
    <div
      class="sc-hKwDye iueJYE"
    >
      에러가 발생하였습니다
    </div>
  </div>
</div>
`;

exports[`<MovieVideo/> Presentational 정상 작동 1`] = `
<div>
  <section
    class="sc-bdvvtL fJWURB"
  >
    <div
      style="width: 100%; height: 100%;"
    />
  </section>
</div>
`;
